app/models/purchase.rb

```diff
@@ -0,0 +1,7 @@
+class Purchase < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :groups, dependent: :destroy
+
+  validates :name, presence: true, length: { maximum: 50 }
+  validates :amount, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```

Comment:
- [ ] It will be great to validate the presence of the amount. This is a major requirement for this project to be approved.

Reply:
@peterrobert, I think you didn't see this because I added a validation test for the amount it the `purchase_test.rb` file.

Reply:
- [ ] This issue has been resolved :heavy_check_mark: